src/QsCompiler/CompilationManager/TypeChecking.cs

```diff
@@ -2015,6 +2015,16 @@ QsCallable GetCallable((QsQualifiedName, ImmutableArray<QsSpecialization>) specI
                 }
                 compilation.UpdateCallables(callables);
                 compilation.UpdateTypes(types);
+
+                var allCallableDeclarations = compilation.GlobalSymbols.DefinedCallables().ToImmutableDictionary(x => x.QualifiedName);
+
+                var callGraph = new SimpleCallGraph(compilation.GetCallables().Values);
+                foreach (var (diag, parent) in callGraph.VerifyAllCycles())
+                {
+                    var info = allCallableDeclarations[parent];
+                    var offset = info.Position is DeclarationHeader.Offset.Defined pos ? pos.Item : null;
+                    diagnostics.Add(Diagnostics.Generate(info.SourceFile.Value, diag, offset));
+                }
```

Comment:
Since this method is already pretty long can we split the cycle diagnostics out into a separate method?

Reply:
done